### PR TITLE
NAS-115373 / 22.12 / NAS-115373: Added checkbox and calls to show data

### PR DESCRIPTION
**Pull request by RehanY147**
To test, try to delete an app after installation.
1. The checkbox to delete unused images should be there
2. If marked checked, you should see another dialog showing the images that can't be deleted because they are still under use.
3. Clicking OK on this dialog should delete the app with the images
4. If not checked, the app should be deleted but the flag `delete_unused_images` should be set to false in the middleware call

### CODE REVIEW

File: src/app/services/dialog.service.ts [hunk 1]
```diff
@@ -77,13 +77,16 @@ export class DialogService {
       dialogRef.componentInstance.method = options.method;
       dialogRef.componentInstance.switchSelectionEmitter.pipe(untilDestroyed(this)).subscribe((selection: boolean) => {
         const data = options.data;
-        if (selection) {
+        if (selection && data[0]) {
```
Comment (AlexKarpov98) [inline src/app/services/dialog.service.ts:80]: Looks weird. It's good to revert it.

**Comment by AlexKarpov98** [comment]
Waiting for success testing

File: src/app/services/dialog.service.ts (hunk 1 above)
Comment (RehanY147) [reply src/app/services/dialog.service.ts:80]: Without the check `data[0]`, I have to pass an object in the dialog config and introduce new ifs into the service with special code just for this PR. This check helps me avoid that scenario. I can just subscribe to the selection change emitter and do my own thing.

File: src/app/services/dialog.service.ts [hunk 2]
```diff
@@ -77,13 +77,16 @@ export class DialogService {
...
           if (data[0] && data[0].hasOwnProperty('reboot')) {
             data[0].reboot = !data[0].reboot;
           }
           if (data[0] && data[0].hasOwnProperty('overcommit')) {
             data[0].overcommit = !data[0].overcommit;
           }
+          if (data[0] && data[0].hasOwnProperty('delete_unused_images')) {
```
Comment (AlexKarpov98) [inline src/app/services/dialog.service.ts:87]: If so, remove data[0] from the next line.

**Review by AlexKarpov98 — APPROVED** [review]

**Comment by RehanY147** [comment]
Needs more work. Can't merge yet.

**Pull request merged**